yarn-project/foundation/src/abi/decoder.ts

```diff
@@ -24,11 +24,30 @@ class AbiDecoder {
     switch (abiType.kind) {
       case 'field':
         return this.getNextField().toBigInt();
-      case 'integer':
+      case 'integer': {
+        const nextField = this.getNextField();
+
         if (abiType.sign === 'signed') {
-          throw new Error('Unsupported type: signed integer');
+          // We get the last (width / 8) bytes where width = bits of type (i64, i32 etc)
```

Comment:
slipping this logic into a function called parse signed integer would mentally simplify this a bit and signify that we are doing two's complement parsing